python-lib/spacy_tokenizer.py

```diff
@@ -0,0 +1,300 @@
+Token.set_extension("is_emoji", getter=lambda token: any(c in UNICODE_EMOJI for c in token.text), force=True)
+Token.set_extension(
+    "is_symbol", getter=lambda token: re.sub(SYMBOL_REGEX, "", token.text) == "", force=True,
+)
+Token.set_extension(
+    "is_unit", getter=lambda token: any([token.lower_.replace(s, "").isdigit() for s in UNITS]), force=True,
+)
+Token.set_extension(
+    "is_time", getter=lambda token: re.sub(TIME_REGEX, "", token.lower_).isdigit(), force=True,
+)
+
+
+class MultilingualTokenizer:
+    """Wrapper class to handle tokenization with spaCy for multiple languages
+
+    Attributes:
+        default_language (str): Fallback language code in ISO 639-1 format
+        use_models (bool): If True, load spaCy models for available languages.
+            Slower but adds additional tagging capabilities to the pipeline.
+        hashtags_as_token (bool): Treat hashtags as one token instead of two
+        batch_size (int): Number of documents to process in spaCy pipelines
+        spacy_nlp_dict (dict): Dictionary holding spaCy Language instances (value) by language code (key)
+        tokenized_column (str): Name of the dataframe column storing tokenized documents
+    """
+
+    DEFAULT_BATCH_SIZE = 1000
+    DEFAULT_NUM_PROCESS = 2
+    DEFAULT_FILTER_TOKEN_ATTRIBUTES = [
+        "is_space",
+        "is_punct",
+        "is_digit",
+        "is_currency",
+        "is_stop",
+        "like_url",
+        "like_email",
+        "like_num",
+        "is_emoji",
+        "is_hashtag",
+        "is_username",
+        "is_symbol",
+        "is_unit",
+        "is_time",
+    ]
+    """list: List of available native and custom spaCy token attributes"""
+
+    def __init__(
+        self,
+        default_language: AnyStr = "xx",  # Multilingual model from spaCy
+        use_models: bool = False,
+        hashtags_as_token: bool = True,
+        batch_size: int = DEFAULT_BATCH_SIZE,
+    ):
+        """Initialization method for the MultilingualTokenizer class, with optional arguments
+
+        Args:
+            default_language (str, optional): Fallback language code in ISO 639-1 format.
+                Default is the "multilingual language code": https://spacy.io/models/xx
+            use_models: If True, loads spaCy models, which is slower but allows to retrieve
+                Part-of-Speech and Entities tags for downstream tasks
+            hashtags_as_token (bool, optional): Treat hashtags as one token instead of two
+                Default is True, which overrides the spaCy default behavior
+            batch_size (int, optional): Number of documents to process in spaCy pipelines
+                Default is set by the DEFAULT_BATCH_SIZE class constant
+        """
+        self.default_language = default_language
+        self.use_models = use_models
+        self.hashtags_as_token = hashtags_as_token
+        self.batch_size = int(batch_size)
+        self.spacy_nlp_dict = {}
```

Comment:
should not write type in variable name

Reply:
I have the same comment as for `column_description_dict`. In this particular case, it's not straightforward that this particular object is a `Dict`. This class is meant to be used by a lot of other classes and other plugins. Even though it is not very elegant, I feel that for this particular object, including dict in the name can avoid confusions from other developers integrating this class.

Reply:
alright !